### ISSUE TITLE
Support Global Setup

### ISSUE THREAD

**Issue by ydcjeff** [vitest-dev/vitest#300]
### Describe the bug

`beforeAll`, `beforeEach`, `afterAll`, and `afterEach` are not running in the expected order if they are in a setup file.

- `beforeAll` is expected to run before **ALL** test file.
- `afterAll` is expected to run after **ALL** test file.

- `beforeEach` is expected to run before **EACH** test file.
- `afterEach` is expected to run after **EACH** test file.

### Reproduction

https://github.com/ydcjeff/vitest-setup-hooks

- pnpm i
- pnpm t

See readme for difference.

### System Info

```shell
System:
    OS: macOS 11.6.2
    CPU: (4) x64 Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
    Memory: 127.75 MB / 8.00 GB
    Shell: 3.3.1 - /usr/local/bin/fish
  Binaries:
    Node: 16.13.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.1.0 - /usr/local/bin/npm
  Browsers:
    Brave Browser: 96.1.33.106
    Chrome: 96.0.4664.110
    Firefox: 95.0
    Firefox Developer Edition: 96.0
    Safari: 15.2
  npmPackages:
    vite: ^2.7.2 => 2.7.6 
    vitest: ^0.0.112 => 0.0.112
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ydcjeff** [comment]
### Additional context

This request comes from migration to Vitest for Vite's tests.

**Comment by antfu** [comment]
Where are the statements from?

Setup files run on **each file** since files are running in isolated env, the subjects of those hooks are always tested cases **NOT test files**. By your definition, there will be no difference between `beforeAll` and `beforeEach`

**Comment by antfu** [comment]
https://jestjs.io/docs/configuration#setupfiles-array

> A list of paths to modules that run some code to configure or set up the testing environment. Each setupFile will be run once per test file. Since every test runs in its own environment.

**Comment by ydcjeff** [comment]
> Where are the statements from?

Those are my request when migrating Vite tests to Vitest locally.

I see. Since tests are running in isolated env, we would need global setup and teardown configurations for launching and shutting down playwright.

**Comment by dominikg** [comment]
discussed this briefly on discord before: https://discord.com/channels/917386801235247114/917386801235247117/921408119312089121

I also thought setupFiles hooks worked the way described in this report. To be able to use vitest for vite, the global hooks are needed unless vite tests get a bit of a refactor. They also use a custom env which is currently not possible in vitest and has some overlap with these global hooks.

TL;DR, we need at least one way to have a shared information context between workers  and global setup/teardown hooks to initialize/cleanup the shared context.
In case of vite that would be starting/stopping the playwright browser and sharing it's websocket address in the context
